src/main/kotlin/org/rust/ide/template/postfix/PrintlnPostfixTemplate.kt

```diff
@@ -0,0 +1,126 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.template.postfix
+
+import com.intellij.codeInsight.template.postfix.templates.PostfixTemplateWithExpressionSelector
+import com.intellij.openapi.editor.Editor
+import com.intellij.psi.PsiElement
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.ext.*
+import org.rust.lang.core.resolve.ImplLookup
+import org.rust.lang.core.resolve.knownItems
+import org.rust.lang.core.types.TraitRef
+import org.rust.lang.core.types.ty.Ty
+import org.rust.lang.core.types.ty.TyReference
+import org.rust.lang.core.types.ty.TyStr
+import org.rust.lang.core.types.ty.TyUnit
+import org.rust.lang.core.types.type
+
+class PrintlnPostfixTemplate(provider: RsPostfixTemplateProvider, private val macroName: String = "println") :
+    PostfixTemplateWithExpressionSelector(null, macroName, "$macroName!(\"{:?}\", expr);",
+        RsTopMostInScopeSelector { it.isNotIgnored && (it.implementsDebug || it.implementsDisplay) }, provider) {
+
+    private class MacroCreator(private val editor: Editor, private val psiFactory: RsPsiFactory,
+                               macroName: String, isDebug: Boolean) {
```

Comment:
Thanks, nice point.
I've slightly modified the example, main points:
* require `Fmt` always and add the `&str` case into `fromExpr` so that it can return an empty formatter also. This allows to throw away the first `if` branch completely and make the code way smaller.

* I removed the trailing `\n` from your snippet, since it seems like they get trimmed from the string and there is no trace of them in the actual expression